### DISCUSSION TITLE
How to disable API option for the UI

### DISCUSSION THREAD

**Discussion by ompraash** [kafbat/kafka-ui#653] [Q&A]
Is there a way to disable the API call. The ask is if the UI is hosted; anybody can call the API and want to disable it. Let the user use the UI for UI purpose and not try to use the API endpoint call.

Is there a toggle option to disable the API call.

**Comment by germanosin** [comment]
Hello @ompraash,

Thank you for your question! The current architecture is built on a REST server and a single-page application. To secure the backend, you can add OAuth, basic authentication, or LDAP authentication. However, disabling REST entirely would disrupt functionality. Instead, you could restrict access to the backend by setting up authentication and limiting requests to specific referrer domains.

I hope this clarifies things! Thanks again for reaching out.

**Reply by Haarolean** [reply]
@ompraash see [this](https://ui.docs.kafbat.io/configuration/authentication) docs category for info on setting up authentication
